plugins/workspace-plugin/src/utils.ts

```
@@ -0,0 +1,94 @@
+                    return container.name;
+                }
+            }
+        }
+    }
+}
+
+export async function getContainerList(): Promise<WorkspaceContainer[]> {
+    const containers: WorkspaceContainer[] = [];
+    try {
+        const workspace = await che.workspace.getCurrentWorkspace();
+
+        if (workspace.runtime && workspace.runtime.machines) {
+            const machines = workspace.runtime.machines;
+            for (const machineName in machines) {
+                if (!machines.hasOwnProperty(machineName)) {
+                    continue;
+                }
+                const container: WorkspaceContainer = { name: machineName, ...machines[machineName] };
+                containers.push(container);
+            }
+        }
+    } catch (e) {
+        throw new Error('Unable to get list workspace containers. Cause: ' + e);
+    }
+    return containers;
+}
+
+export async function execInTerminal(directory: string, locationURI: string, ...args: string[]): Promise<void> {
+    const editorContainerName = await getEditorContainerName() as string;
+    try {
+        const shellArgs: string[] = ['-c', 'git'].concat(args);
+        const terminalOptions: theia.TerminalOptions = {
+            cwd: directory,
+            shellPath: 'sh',
+            shellArgs: shellArgs,
+            name: `${args[0]} ${locationURI}`,
+            attributes: {
+                CHE_MACHINE_NAME: editorContainerName,
+                closeWidgetExitOrError: 'false',
+                interruptProcessOnClose: 'true'
+            }
+        };
+        const terminal = theia.window.createTerminal(terminalOptions);
+        terminal.show();
+        await terminal.processId;
+    } catch (error) {
+        console.error(`Couldn't clone ${locationURI}: ${error.message}`);
+        throw new Error(`Couldn't clone ${locationURI}: ${error.message}`);
+    }
+}
+
+export async function checkRepoAccess(locationURI: string): Promise<boolean> {
```

If you test for 401, then the name of the method should be changed something like `isRepositoryPrivate`.
Btw, network errors are not handled here. The method will always return false for public repository and when error.